python/freetensor/core/remote_task_scheduler.py

```diff
@@ -0,0 +1,735 @@
+            threading.Thread(target=self.task_submit, args=(server_uid,))
+
+    def update_inavailability(self, server_uid: str, time_stamp: float):
+        self.server_list_lock.acquire()
+        if self.server_list[server_uid][1] < time_stamp:
+            self.server_list[server_uid] = (self.server_list[server_uid][1],
+                                            time_stamp)
+            self.available_server_list.add(server_uid)
+        self.server_list_lock.release()
+
+    def task_execution(self, task: Task):
+        tmprunner = TaskRunner(task, self)
+        self.execution_queue_cnt_lock.acquire()
+        self.execution_queue_cnt += 1
+        self.execution_queue_cnt_lock.release()
+        tmprunner.start()
+
+    def remote_result_receive(self, remote_host_uid: str,
+                              task_result: TaskResult):
+        if task_result["task_type"] == 1:
+            pass
+        elif task_result["task_type"] == 2:
+            tmptask_result = MeasureResult.dict2self(task_result)
+        self.task_merge(tmptask_result)
+
+    def remote_logger(self, log: str):
+        pass
+
+    def remote_logger_execution(self, dict):
+        pass
+
+    def get_queue_len(self) -> Dict[int, int]:
+        return {
+            "0": self.execution_queue_cnt,
+            "1": self.search_queue.qsize(),
+            "2": self.measure_queue.qsize()
+        }
+
+    def add_host(self, _server_uid: str, sev_status: int) -> None:
+        self.server_list_lock.acquire()
+        self.server_list[_server_uid] = (sev_status, time.time())
+        if sev_status == 1:
+            self.search_server_num += 1
+        if sev_status == 2:
+            self.measure_server_num += 1
+        if sev_status == 3:
+            self.measure_server_num += 1
+            self.search_server_num += 1
+        if (self.measure_server_num + self.search_server_num) < 3:
+            self.report_new_task()
+        self.server_list_lock.release()
+
+    def remove_host(self, server_uid: str) -> None:
+        self.server_list_lock.acquire()
+        sev_status = self.server_list.pop(server_uid)[0]
+        if sev_status == 1:
+            self.search_server_num -= 1
+        if sev_status == 2:
+            self.measure_server_num -= 1
+        if sev_status == 3:
+            self.measure_server_num -= 1
+            self.search_server_num -= 1
+        self.server_list_lock.release()
+        self.submitted_task_container_lock.acquire()
+        for key in self.submitted_task_container.keys():
+            if self.submitted_task_container[
+                    key].target_server_uid == server_uid:
+                self.task_trans_submit2waiting(key)
+        self.submitted_task_container_lock.release()
+
+    def get_self_uid(self, server_uid: str) -> None:
+        self.self_server_uid = server_uid
+        if self.is_ready == False:
+            self.submit_queue_lock.release()
+
+
+#the followings are functional tests
```

Comment:
Please move the tests to the `test` directory

Reply:
Ok, I will rewrite the tests then